### PR TITLE
#161966918 User Comments

### PR DESCRIPTION
**Pull request by leonardnjura**
#### What does this PR do?
- Integrates user ability to leave comments on articles and threads
- Allows comment owner to edit or delete comments

#### How should this be manually tested?
`$ git clone -b ft-user-comments-161966918 https://github.com/andela/ah-infinity-stones`
`$ cd ah-infinity-stones`
> Create and activate a virtual envirinment to hold this app's dependancies
`$ virtualenv -p python3 venv`
`$ source venv/bin/activate`
`$ pip3 install -r requirements.txt to install dependencies`
> Setup environment variables in a file called `.env` with your own values. See sample below
> Load the `.env` variables by `$ source .env`
> Prepare migrations: `$ python3 manage.py makemigrations` and
> Run migrations: `$ python3 manage.py migrate`
> Run generic tests `$ python3 manage.py test`
> Run specific tests `$ python3 manage.py test authors/apps/comments`
> Register for a token to use on protected endpoint unless Read Only
> Engage the following endpoints:

> `POST http://127.0.0.1:8000/api/users/` to register
> `POST http://127.0.0.1:8000/api/articles/<slug>/comments`
> `GET http://127.0.0.1:8000/api/articles/<slug>/`
> `GET PUT DELETE http://127.0.0.1:8000/api/articles/<slug>/comments/1`

Populate your `.env` with:
```
export DATABASE_URL=postgres://<database_user>:<database_password>@localhost/<database_name>

 export EMAIL_HOST_USER='sendgrid_username'
 export EMAIL_HOST_PASSWORD='sendgrid_pswd'
 export SECRET_KEY='your_secret'
```

#### What are the relevant pivotal tracker stories?
[#161966918](https://www.pivotaltracker.com/story/show/161966918)

#### Screenshots
screen shot 2019-01-08 at 10 05 49

screen shot 2019-01-08 at 10 08 23

screen shot 2019-01-08 at 10 09 14

screen shot 2019-01-08 at 10 09 23

### CODE REVIEW

**Comment by codeplus254** [comment]
Great work on the comments aspect of Authors Haven!

**Comment by PromasterGuru** [comment]
Great work, kindly resolve the merge conflicts 👍

**Comment by ThaDeveloper** [comment]
Kindly confirm the endpoint URL format for comments from the README

**Review by PromasterGuru — APPROVED** [review]

File: authors/apps/articles/urls.py [hunk 1]
```diff
@@ -23,9 +23,6 @@
         ArticleLikeDislikeView.as_view(),
         name='like_article'),
     path('articles/search/', SearchArticleView.as_view(), name='search'),
-    path('comment/', CommentCreateViewAPIView.as_view(), name='comment'),
```
Comment (codeplus254) [inline authors/apps/articles/urls.py:26]: You no longer need this routes?

**Review by kamyaD — APPROVED** [review]

**Review by codeplus254 — APPROVED** [review]
Great work

**Pull request merged**